From b1fea1a6c482ee87d862937293a1a0cebedeaf04 Mon Sep 17 00:00:00 2001
From: Allon Huang <allon.huang@rock-chips.com>
Date: Wed, 29 Jul 2020 20:55:18 +0800
Subject: [PATCH] media: rockchip: cif: fix lost frame when sof occurs twice at
 the same time

frm start int don't be set, current & next frame start occur simultaneously

Signed-off-by: Allon Huang <allon.huang@rock-chips.com>
Change-Id: I49f28a6d49e8a726f3f8a4c290444792a497876f
---
 drivers/media/platform/rockchip/cif/capture.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/rockchip/cif/capture.c b/drivers/media/platform/rockchip/cif/capture.c
index d328f04a1e25..d4898b331c76 100644
--- a/drivers/media/platform/rockchip/cif/capture.c
+++ b/drivers/media/platform/rockchip/cif/capture.c
@@ -1353,9 +1353,8 @@ static int rkcif_csi_channel_set(struct rkcif_stream *stream,
 				 CSI_DMA_END_INTSTAT(channel->id) |
 				 CSI_LINE_INTSTAT(channel->id)));
 
-	/* enable mipi id0 frame start int for sof(long frame) */
-	if (dev->hdr.mode != NO_HDR &&
-	    channel->id == RKCIF_STREAM_MIPI_ID0)
+	/* enable id0 frame start int for sof(long frame, for hdr) */
+	if (channel->id == RKCIF_STREAM_MIPI_ID0)
 		rkcif_write_register_or(dev, CIF_REG_MIPI_LVDS_INTEN,
 					CSI_START_INTEN(channel->id));
 
-- 
2.35.3

